azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/cache/CacheableAspect.java

```diff
@@ -0,0 +1,81 @@
+package com.microsoft.azure.toolkit.lib.common.cache;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.microsoft.azure.toolkit.lib.common.utils.aspect.ExpressionUtils;
+import com.microsoft.azure.toolkit.lib.common.utils.aspect.MethodInvocation;
+import lombok.extern.java.Log;
+import org.apache.commons.lang3.StringUtils;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Pointcut;
+import org.aspectj.lang.reflect.MethodSignature;
+
+import javax.annotation.Nonnull;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+@Aspect
+@Log
+public class CacheableAspect {
+    private static final CacheLoader<String, Cache<Object, Object>> loader = new CacheLoader<String, Cache<Object, Object>>() {
+        @Override
+        public Cache<Object, Object> load(@Nonnull String key) {
+            return CacheBuilder.newBuilder()
+                    .softValues()
+                    .expireAfterAccess(4, TimeUnit.HOURS) // TODO: justify
```

Comment:
use a constant default expiration

Reply:
the key is never used in load?

Reply:
yes, we always create a new cache container.